controllers/tasks.js

```diff
@@ -401,6 +401,27 @@ const assignTask = async (req, res) => {
   }
 };
 
+const updateOldTaskStatus = async (req, res) => {
+  try {
+    const allTasks = await tasks.fetchTasks();
+    const allOldTasks = allTasks.filter(
+      (task) => task.status.toUpperCase() === "COMPLETED" || task.status === "unassigned"
+    );
+    const updatedTasks = [];
+    for (const task of allOldTasks) {
+      if (task.status.toUpperCase() === "COMPLETED") {
+        updatedTasks.push(await tasks.updateTask({ status: "DONE" }, task.id));
+      } else if (task.status === "unassigned") {
+        updatedTasks.push(await tasks.updateTask({ status: "UNASSIGNED" }, task.id));
+      }
+    }
+    return res.json({ message: `Updated Old tasks`, tasks: updatedTasks });
+  } catch (error) {
+    logger.error(`Error while Updating tasks ${error}`);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
```

Comment:
This is business logic. Should ideally happen in model file instead of controller